lib/ansible/plugins/inventory/hcloud.py

```diff
@@ -0,0 +1,200 @@
+        if len(self.get_option("label_selector")) > 0:
+            self.servers = self.client.servers.get_all(label_selector=self.get_option("label_selector"))
+        else:
+            self.servers = self.client.servers.get_all()
+
+    def _filter_servers(self):
+        if len(self.get_option("locations")) > 0:
+            tmp = []
+            for server in self.servers:
+                if server.datacenter.location.name in self.get_option("locations"):
+                    tmp.append(server)
+            self.servers = tmp
+
+        if len(self.get_option("types")) > 0:
+            tmp = []
+            for server in self.servers:
+                if server.server_type.name in self.get_option("types"):
+                    tmp.append(server)
+            self.servers = tmp
+
+        if len(self.get_option("images")) > 0:
+            tmp = []
+            for server in self.servers:
+                if server.image is not None and server.image.name in self.get_option("images"):
+                    tmp.append(server)
+            self.servers = tmp
+
+    def _set_server_attributes(self, server):
+        self.inventory.set_variable(server.name, "id", to_native(server.id))
+        self.inventory.set_variable(server.name, "name", to_native(server.name))
+        self.inventory.set_variable(server.name, "status", to_native(server.status))
+
+        # Network
+        self.inventory.set_variable(server.name, "ipv4", to_native(server.public_net.ipv4.ip))
+        self.inventory.set_variable(server.name, "ipv6_network", to_native(server.public_net.ipv6.network))
+        self.inventory.set_variable(server.name, "ipv6_network_mask", to_native(server.public_net.ipv6.network_mask))
+
+        if self.get_option("connect_with") == "public_ipv4":
+            self.inventory.set_variable(server.name, "ansible_host", to_native(server.public_net.ipv4.ip))
+        elif self.get_option("connect_with") == "hostname":
+            self.inventory.set_variable(server.name, "ansible_host", to_native(server.name))
+        elif self.get_option("connect_with") == "ipv4_dns_ptr":
+            self.inventory.set_variable(server.name, "ansible_host", to_native(server.public_net.ipv4.dns_ptr))
+
+        # Server Type
+        self.inventory.set_variable(server.name, "server_type", to_native(server.image.name))
+
+        # Datacenter
+        self.inventory.set_variable(server.name, "datacenter", to_native(server.datacenter.name))
+        self.inventory.set_variable(server.name, "location", to_native(server.datacenter.location.name))
+
+        # Image
+        self.inventory.set_variable(server.name, "image_id", to_native(server.image.id))
+        self.inventory.set_variable(server.name, "image_name", to_native(server.image.name))
+
+    def verify_file(self, path):
+        """ return true/false if this is possibly a valid file for this plugin to consume """
+        valid = False
+        if super(InventoryModule, self).verify_file(path):
```

Comment:
Even better, this whole thing can be rewritten as:
```suggestion
        return (
            super(InventoryModule, self).verify_file(path) and
            path.endswith((self.NAME + ".yaml", self.NAME + ".yml"))
        )
```

Reply:
Currently it is exactly the sample from the docs:
https://docs.ansible.com/ansible/devel/dev_guide/developing_inventory.html#inventory-plugins

But i think this way is better